runtime/dgraphclient/registry.go

```diff
@@ -65,13 +65,6 @@ func ShutdownConns() {
 }
 
 func (dr *dgraphRegistry) getDgraphConnector(ctx context.Context, dgName string) (*dgraphConnector, error) {
-	dr.RLock()
-	ds, ok := dr.dgraphConnectorCache[dgName]
-	dr.RUnlock()
-	if ok {
-		return ds, nil
-	}
-
```

Comment:
To elaborate, when a resource is protected by a reader/writer lock (an `sync.RWMutex` in Go), a function that is primarily concerned with retrieving the resource should obtain a read lock, not a full write lock.

In this case, the resource is cached, so in the case of a cache miss, the read lock has to be released, and then a new write lock can be obtained (other languages have a "promote" concept to avoid a true release, but Go does not).  Since the cache may have been updated by another thread while there was no lock, the first thing the new full lock has to try is once again reading from the cache.  Only in the case of a second cache miss while holding a full write lock, does it then attempt to create the resource and update the cache.

This is similar to "double-checked locking" in other languages.

Reply:
Researching this a bit more... while it's not incorrect to do the double-checked lock approach, there's a built in that handles it for you that I didn't know about - [`sync.Map`](https://pkg.go.dev/sync#Map).  Perhaps switch us to that?

more: https://victoriametrics.com/blog/go-sync-map/

Reply:
... and even more research - `sync.Map` still has problems.  The main one is that unlike a regular map - it isn't generic but instead uses `any` for both key and value types.

Digging further, It would seem the best option for a performant generic map is currently `xsync.MapOf[K, V]` from [this library](https://github.com/puzpuzpuz/xsync). Want to give it a try?

Reply:
I asked claude and it said double locking is unnecessary and the minimal benefits aren’t really worth it. Getting the full write lock isn’t that much worse than just the read lock. I got confused reading it which says something imo

Reply:
Per discussion, let's go back to how it was, and I will address in a separate PR.  Thanks.